### PR TITLE
Add a heretics sacrifice targets to the end of round report

### PR DESCRIPTION
**Pull request by Jackal-boop**
## About The Pull Request
adds a heretics sac targets to the end of round report
## Why It's Good For The Game
lets others know who heretics needed to sac
image

## Changelog
:cl:
qol: heretic sac targets now appear on the end of round report
/:cl:

### CODE REVIEW

**Review by MrMelbert — COMMENTED** [review]
Notably this will only report all incomplete sacrifice targets they had at the end of the round - it won't count re-rolled targets or successfully complete targets

That being said it doesn't seem extremely useful in this state.

Can you provide a picture of what it looks like, just to showcase the formatting?

**Comment by Jackal-boop** [comment]
> Notably this will only report all incomplete sacrifice targets they had at the end of the round - it won't count re-rolled targets or successfully complete targets
> 
> That being said it doesn't seem extremely useful in this state.
> 
> Can you provide a picture of what it looks like, just to showcase the formatting?

surre ill post a picture later when i can

**Comment by Jackal-boop** [comment]
how could i show completed targets and rerolled targets?

**Comment by MrMelbert** [comment]
> how could i show completed targets and rerolled targets?

Personally I would only show completed sacrifices but if you really want to show both, simply keep a list on the heretic datum and, when a sacrifice target is added, add their `real_name` to the list

IE (pseudo code ahead)
```dm
/datum/heretic
   var/list/all_sacs = list()
```
```dm
/datum/heretic/proc/add_sacrifice_target()
   ...
   all_sacs += target.real_name
```

**Comment by Jackal-boop** [comment]
i think i took on something a bit outside of my abilities, hey @Rhials you could do this tho :3c

**Comment by Jackal-boop** [comment]
i cannot for the life of me find where sacrificing a person is actually handeled

**Comment by Rhials** [comment]
> i think i took on something a bit outside of my abilities, hey @Rhials you could do this tho :3c

You can do it!!!!! Don't give up!!!!!!!

I can provide more than moral support if you want, but I totally think this something you can handle

**Comment by Jackal-boop** [comment]
> > i think i took on something a bit outside of my abilities, hey @Rhials you could do this tho :3c
> 
> You can do it!!!!! Don't give up!!!!!!!
> 
> I can provide more than moral support if you want, but I totally think this something you can handle

alright ill take another crack at it this week, bit busy though so may take a few days before i can really sit down and work on it

File: code/modules/antagonists/heretic/heretic_antag.dm [hunk 1]
```diff
@@ -23,6 +23,7 @@
 	hijack_speed = 0.5
 	suicide_cry = "THE MANSUS SMILES UPON ME!!"
 	preview_outfit = /datum/outfit/heretic
+	var/list/all_sac_targets = list()
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/heretic_antag.dm:26]: I'd ask that you put this down at `var/list/mob/living/carbon/human/sac_targets`
Also auto-doc

File: code/modules/antagonists/heretic/heretic_antag.dm [hunk 2]
```diff
@@ -408,6 +409,7 @@
 
 	LAZYSET(sac_targets, target, target_image)
 	RegisterSignal(target, COMSIG_QDELETING, PROC_REF(on_target_deleted))
+	all_sac_targets.Add(target.real_name)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/heretic_antag.dm:412]:
```suggestion
	all_sac_targets += target.real_name
```

File: code/modules/antagonists/heretic/heretic_antag.dm [hunk 3]
```diff
@@ -447,7 +449,9 @@
 
 	parts += printplayer(owner)
 	parts += "<b>Sacrifices Made:</b> [total_sacrifices]"
-
+	var/roundend_targets = jointext(all_sac_targets, ", ")
+	var/targets = "The heretics targets were [roundend_targets]"
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/heretic_antag.dm:453]:
```suggestion
	var/roundend_targets = jointext(all_sac_targets, ", ")
	var/targets = "The heretic's targets were: [roundend_targets]."
```
Though also I suggest
```suggestion
	var/targets = "The heretic's targets were: [english_list(all_sac_targets, nothing_text = "No one")]."
```

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Still needs pictures

File: code/modules/antagonists/heretic/heretic_antag.dm (hunk 1 above)
Comment (Jackal-boop) [reply code/modules/antagonists/heretic/heretic_antag.dm:26]: whats auto doc

File: code/modules/antagonists/heretic/heretic_antag.dm (hunk 3 above)
Comment (Jackal-boop) [reply code/modules/antagonists/heretic/heretic_antag.dm:453]: yeah thats a good idea

**Comment by Jackal-boop** [comment]
image aadded

**Comment by Jackal-boop** [comment]
cat: ci_test/data/logs/ci/clean_run.lk: No such file or directory
Error: Process completed with exit code 1.

why was this error raised?

File: code/modules/antagonists/heretic/heretic_antag.dm [hunk 4]
```diff
@@ -43,6 +43,8 @@
 	var/high_value_sacrifices = 0
 	/// Lazy assoc list of [refs to humans] to [image previews of the human]. Humans that we have as sacrifice targets.
 	var/list/mob/living/carbon/human/sac_targets
+	// list of all targets used for end of round report
+	var/list/all_sac_targets = list()
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/heretic_antag.dm:47]:
```suggestion
	/// List of all sacrifice target's names, used for end of round report
	var/list/all_sac_targets = list()
```

File: code/modules/antagonists/heretic/heretic_antag.dm [hunk 5]
```diff
@@ -447,7 +450,8 @@
 
 	parts += printplayer(owner)
 	parts += "<b>Sacrifices Made:</b> [total_sacrifices]"
-
+	var/targets = "The heretic's targets were: [english_list(all_sac_targets, nothing_text = "No one")]."
+	parts += targets
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/heretic_antag.dm:454]:
```suggestion
	parts += "The heretic's targets were: [english_list(all_sac_targets, nothing_text = "No one")]."
```

**Pull request merged**